samples/charts/financial-chart/axis-types/Services/StocksUtility.cs

```diff
@@ -0,0 +1,210 @@
+                    Low = l,
+                    Close = c,
+                    Volume = v };
+
+                stock.Add(stockItem);
+
+                o = c + ((random.NextDouble() - 0.5) * priceRange);
+                if (o < 0)
+                {
+                    o = Math.Abs(o) + 2;
+                }
+                h = o + (random.NextDouble() * priceRange);
+                l = o - (random.NextDouble() * priceRange);
+                c = l + (random.NextDouble() * (h - l));
+                v = v + ((random.NextDouble() - 0.5) * volumeRange);
+                if (v < 0)
+                {
+                    v = Math.Abs(v) + 10000;
+                }
+
+                o = Math.Round(o * 100) / 100;
+                h = Math.Round(h * 100) / 100;
+                l = Math.Round(l * 100) / 100;
+                c = Math.Round(c * 100) / 100;
+                v = Math.Round(v * 100) / 100;
+
+                time = time.AddMinutes(interval);
+            }
+            //setting data intent for Series Title
+            //   (stock as any).__dataIntents = {
+            //    close: ["SeriesTitle/Stock Prices"]
+            //};
+
+            return stock;
+        }
+
+        public static string toShortString(double largeValue)
+        {
+            double roundValue;
+
+            if (largeValue >= 1000000)
+            {
+                roundValue = Math.Round(largeValue / 100000) / 10;
+                return roundValue + "M";
+            }
+            if (largeValue >= 1000)
+            {
+                roundValue = Math.Round(largeValue / 100) / 10;
+                return roundValue + "K";
+            }
+
+            roundValue = Math.Round(largeValue);
+            return roundValue + "";
+        }
+
+        public static DateTime GetYear(DateTime date)
+        {
+            return new DateTime(date.Year);
+        }
+
+        public static double GetQuarter(DateTime date)
+        {
+            var month = date.Month;
+            double d = (month + 2) / 3;
+            return Math.Round(d);
+        }
+
+        public static StockItem GetLastItem(List<StockItem> array)
+        {
+            if (array.Count == 0)
+            {
+                return null;
+            }
+            return array.LastOrDefault();
+        }
+
+        public static List<StockItem> GetNewItem(List<StockItem> array, int ?interval)
+        {
+            var lastItem = GetLastItem(array);
+
+            if (interval == null)
+            {
+                interval = intervalDays * 24 * 60;
+                interval += intervalHours * 60;
+                interval += intervalMinutes;
+            }
+
+            var time = lastItem.Date.AddMinutes(interval.Value);
+            var v = lastItem.Volume;
+            var o = lastItem.Open;
+            var h = lastItem.High;
+            var l = lastItem.Low;
+            var c = lastItem.Close;
+
+            Random random = new Random();
```

Comment:
the random variable should be a static field in this class instead of local variable of this function

Reply:
make sure you make the same corrections in StocksUtility.cs in other samples

Reply:
pending